Code/GraphMol/Descriptors/RDF.cpp

```diff
@@ -0,0 +1,350 @@
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Institue of Cancer Research.
+//       nor the names of its contributors may be used to endorse or promote
+//       products derived from this software without specific prior written
+//       permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// For more information on the Plane of Best Fit please see
+// http://pubs.acs.org/doi/abs/10.1021/ci300293f
+//
+//  If this code has been useful to you, please include the reference
+//  in any work which has made use of it:
+
+//  Plane of Best Fit: A Novel Method to Characterize the Three-Dimensionality
+//  of Molecules, Nicholas C. Firth, Nathan Brown, and Julian Blagg, Journal of
+//  Chemical Information and Modeling 2012 52 (10), 2516-2525
+
+//
+//
+// Created by Nicholas Firth, November 2011
+// Modified by Greg Landrum for inclusion in the RDKit distribution November
+// 2012
+// Further modified by Greg Landrum for inclusion in the RDKit core September
+// 2016
+// Adding RBF descriptors to 3D descriptors by Guillaume Godin
+
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/MolOps.h>
+#include <GraphMol/PeriodicTable.h>
+
+#include <GraphMol/MolTransforms/MolTransforms.h>
+
+#include "RDF.h"
+#include "MolData3Ddescriptors.h"
+
+#include <Numerics/Matrix.h>
+#include <Numerics/SquareMatrix.h>
+#include <Numerics/SymmMatrix.h>
+#include <boost/foreach.hpp>
+#include <math.h>
+#include <Eigen/Dense>
+
+// data checked using book Todeschini R., Consonni V. - Molecular Descriptors for Chemoinformatics 2009 atomic properties page 21/22
+
+
+
+namespace RDKit {
+namespace Descriptors {
+namespace {
+
+  MolData3Ddescriptors moldata3D;
+
+
+std::vector<double> getG(int n) {
+  std::vector<double> res(n);
+  for (int i = 0; i < n; i++) {
+    res[i] = 1 + i * 0.5;
+  }
+  return res;
+}
```

Comment:
In Python I would write something like this:
```
def getSequence(length, start=0, step=1):
  return [start + i*step for i in range(length)]
```
That would implement the other `getG` function above as default.